src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -33,6 +35,15 @@ class StorageExternalMixin(ExternalMixin):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+    @property
+    def filtered_bids_next_market(self) -> List[Dict]:
+        """Get a representation of each of the device's bids from the next market."""
+
+        return [
+            {'id': bid.id, 'price': bid.price, 'energy': bid.energy}
+            for _, bid in self.next_market.get_bids().items()
+            if bid.buyer == self.device.name]
+
```

Comment:
IMO it would be great to add this property in the `ExternalMixin` class, as the users of it are direct children of it.
I would also vote for making it a bit more generic ( if it can return either bids or offers )

Reply:
It's true that all the classes that use `filtered_bids_next_market` inherit from `ExternalMixin`. However, not all classes that inherit from `ExternalMixin` use `filtered_bids_next_market`.

- `LoadExternalMixin`
- `PVExternalMixin`  # This class doesn't make bids
- `StorageExternalMixin`

The same applies for the offers.
I agree that having this duplication isn't nice nor DRY. But I don't think that adding it to `ExternalMixin` would be correct (since it doesn't apply to all the children classes). In theory, this would probably belong to a new class (something like `ExternalBidMixin`) but it would probably be a bit too much for one method. I'm not sure, to be honest.

Reply:
I think we have discussed it again, but we were under the same dilemma when implementing these classes. The most straightforward way to do this is to implement separate ExternalBidMixin and ExternalOfferMixin mixins, as @fievelk suggested. However, this would bloat our class hierarchy, because the class hierarchy tree would become more complex. Therefore we decided to bite the bullet and do the code a bit less DRY than to complicate our class hierarchy further. 
Having said that, we can revisit it in the future and see if it makes sense now to create such mixins. Not in the content of this PR though.